tests/phpunit/api/v3/PaymentProcessorTest.php

```diff
@@ -104,7 +115,24 @@ public function testPaymentProcessorUpdate() {
       'payment_type' => 1,
       'payment_instrument_id' => 1,
       'is_active' => 1,
+      'user_name' => NULL,
+      'password' => NULL,
+      'signature' => NULL,
+      'url_site' => NULL,
+      'url_api' => NULL,
+      'url_recur' => NULL,
+      'url_button' => NULL,
+      'subject' => NULL,
+      'accepted_credit_cards' => NULL,
     ];
+    if ($version === 3) {
+      // In APIv3 If a field is default NULL it is not returned.
```

Comment:
@colemanw not sure if this is deliberate but it seems a noticeable change between the API versions, i assume it is documented somewhere?

Reply:
Yes it is deliberate.
https://github.com/civicrm/org.civicrm.api4/pull/116

You could simplify this patch by deleting NULL values from the api result instead of adding them to the expected result :)

Reply:
ok have updated based on your suggestion @colemanw